workspace/data/rules_py/acme_rtu-01_fan_and_system_run_hours.py

```
@@ -40,88 +17,73 @@ def _system_on_series(df, cfg, fan_on):
+    return pa.array(out, type=pa.float64())
 
 
 def _sum_hours(mask, dt_hours):
-    return float(dt_hours.loc[mask.fillna(False)].sum())
+    pairs = zip(mask.to_pylist(), dt_hours.to_pylist(), strict=False)
+    return round(sum(h for m, h in pairs if m and h is not None), 3)
 
 
-# Script entry — executed by Rule Lab / FDD batch runner (mode=script).
-ts_col = "timestamp" if "timestamp" in df.columns else ("ts" if "ts" in df.columns else None)
+ts_col = "timestamp" if "timestamp" in table.column_names else ("ts" if "ts" in table.column_names else None)
```

Ruff (0.15.15)

[error] Undefined name `table`

(F821)